### PR TITLE
Added random choice intial commit

### LINKED ISSUE #23259: choice

### PR DESCRIPTION
**Pull request by rsaeta**
# PR Description
Add frontend for `jax.random.choice`

## Related Issue

Close #23259

## Checklist

- [x] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you follow the steps we provided?

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_jax/test_random.py [hunk 1]
```diff
@@ -408,6 +408,96 @@ def call():
...
+
+    if not ivy.exists(ret):
+        return
+
+    ret_np, ret_from_np = ret
+    ret_np = helpers.flatten_and_to_np(ret=ret_np, backend=backend_fw)
+    ret_from_np = helpers.flatten_and_to_np(ret=ret_from_np, backend=backend_fw)
+    for u, v in zip(ret_np, ret_from_np):
+        assert u.dtype == v.dtype
+
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_random.py:499]: Hi @rsaeta, could you also assert the shapes are the same? Other than that, looks good to me!

**Review by AnnaTz — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
